app/scripts/metamask-controller.test.js

```diff
@@ -742,257 +802,283 @@ describe('MetaMaskController', () => {
-        jest.spyOn(window, 'open').mockReturnValue();
+        [HardwareDeviceNames.trezor, HardwareDeviceNames.ledger].forEach(
+          (device) => {
+            describe(`using ${device}`, () => {
+              it('should be unlocked by default', async () => {
+                await metamaskController.connectHardware(device, 0);
 
-        const localMetaMaskController = new MetaMaskController({
-          showUserConfirmation: noop,
-          encryptor: mockEncryptor,
-          initState: {
-            ...cloneDeep(firstTimeState),
-            KeyringController: {
-              keyrings: [{ type: KeyringType.trezor, accounts: ['0x123'] }],
-              isUnlocked: true,
-            },
-            PreferencesController: {
-              identities: {
-                '0x123': { name: 'Trezor 1', address: '0x123' },
-              },
-              selectedAddress: '0x123',
-            },
-          },
-          initLangCode: 'en_US',
-          platform: {
-            showTransactionNotification: () => undefined,
-            getVersion: () => 'foo',
+                const status = await metamaskController.checkHardwareStatus(
+                  device,
+                );
+
+                expect(status).toStrictEqual(true);
+              });
+            });
           },
-          browser: browserPolyfillMock,
-          infuraProjectId: 'foo',
-          isFirstMetaMaskControllerSetup: true,
+        );
+      });
+
+      describe('forgetDevice', () => {
+        it('should throw if it receives an unknown device name', async () => {
+          const result = metamaskController.forgetDevice(
+            'Some random device name',
+          );
+          await expect(result).rejects.toThrow(
+            'MetamaskController:getKeyringForDevice - Unknown device',
+          );
         });
 
-        await localMetaMaskController.keyringController.createNewVaultAndKeychain(
-          'password',
-        );
+        it('should remove the identities when the device is forgotten', async () => {
+          await metamaskController.connectHardware(
+            HardwareDeviceNames.trezor,
+            0,
+          );
+          await metamaskController.unlockHardwareWalletAccount(
+            0,
+            HardwareDeviceNames.trezor,
+          );
+          const hardwareKeyringAccount =
+            metamaskController.keyringController.state.keyrings[1].accounts[0];
 
-        await localMetaMaskController.keyringController.addNewKeyring(
-          'Trezor Hardware',
-          {
-            accounts: ['0x123'],
-          },
-        );
+          await metamaskController.forgetDevice(HardwareDeviceNames.trezor);
 
-        await localMetaMaskController.forgetDevice(HardwareDeviceNames.trezor);
-        const { identities: updatedIdentities } =
-          localMetaMaskController.preferencesController.store.getState();
-        expect(updatedIdentities['0x123']).toBeUndefined();
-      });
+          expect(
```

Comment:
We should also check that its gone in the accounts controller

Reply:
Good point! added in 42788a75897906b5a8a3068727ae04f62454ed38

Reply:
I also noticed that the assertion related to PreferencesController was broken, I fixed it in 9e16552a43eacdbc9267813b5ff06157d73a2de2